frontend/src/hooks/fetchUser.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react'
+import axios from 'axios'
+
+export const fetchUser = (): any => {
+  const [user, setUser] = useState<any>(null)
+  const [loadingStatus, setLoadingStatus] = useState<string>('unloaded')
+
+  useEffect(() => {
+    const getUserData = async (): Promise<void> => {
+      try {
+        setLoadingStatus('loading')
+        const userDataResponse = await axios.get('http://localhost:8000/userData/1', { withCredentials: true })
+        setUser(userDataResponse.data)
+        setLoadingStatus('loaded')
+      } catch (error) {
+        setLoadingStatus('error')
+        console.error(error)
+      }
+    }
+    void getUserData()
```

Comment:
i don't know if the catch in the getUserData gets thrown correctly if this line 20 here doesn't also include a .catch block

Reply:
yes, the catch block caught the errors when tested it. 

perhaps this was something you're looking for?

![image](https://user-images.githubusercontent.com/106122976/233270384-7306fad4-889c-4ef1-94d3-62248864a3c0.png)